### PR TITLE
Add scroll to top in scroll window #342

### LINKED ISSUE #342: Add scroll to top in scroll window
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/susi_server

**Current behavior:**
Only scroll to bottom present as of now

**Expected behavior:**
Scroll to top to see previous chats

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #342 

Closed PR https://github.com/fossasia/susi_chromebot/pull/356 due to some merge issues.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Added scroll to top functionality

Screenshot
New
scroll-up-down

Old
screen shot 2018-09-24 at 6 45 21 pm

#### Changes proposed in this pull request:
 - Scroll to top
 - Code indentation
 - Removed redundant new lines

### CODE REVIEW

**Comment by ms10398** [comment]
@fragm3 Does a scroll to top button really makes sense with scroll to bottom button 

What are your views about it @gabru-md

**Comment by fragm3** [comment]
@ms10398 on the susi https://chat.susi.ai, its present. Since the message window is small, it make sense according to me

**Comment by zamhaq** [comment]
> @ms10398 on the susi [chat.susi.ai](https://chat.susi.ai), its present. Since the message window is small, it make sense according to me

> @ms10398 on the susi chat.susi.ai, its present. Since the message window is small, it make sense according to me

@fragm3 both buttons together are blocking a lot of space in this small view. If we need it, can we make it so that both the buttons appear on the same place i.e `lower right` depending on if user is at the `bottom` or the `top`?

**Comment by ms10398** [comment]
Yeah as it felt wierd having two buttons diagonally in a small window

**Comment by fragm3** [comment]
@zamhaq sounds good. What about when scroll is in between i.e. both up and down are present, which scroll should be visible to user in this case(at lower right positon)

**Comment by rupav** [comment]
@fragm3 when the user is scrolling downwards, show them "scroll to bottom". When the user is scrolling up, show then the other one. On window.onload, i.e. in default, show them "scroll to top".

**Comment by fragm3** [comment]
@rupav sounds good, working on it!

File: src/popup.html [hunk 1]
```diff
@@ -52,7 +52,10 @@
             <div class="messages" id="messages">
             </div>
         </div>
-        <span style="z-index: 1;" class="scroll-icon" id="scrollIcon">
+        <span class="scroll-icon-top" id="scrollIconTop">
+            <svg viewBox="0 0 24 24" style="display: inline-block; color: rgb(255, 255, 255); fill: rgb(255, 255, 255); height: 30px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; line-height: 40px;"><path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"></path></svg>
+        </span>
+        <span style="z-index: 1;" class="scroll-icon-bottom" id="scrollIconBottom">
         <svg viewBox="0 0 24 24" style="display: inline-block; color: rgb(255, 255, 255); fill: rgb(255, 255, 255); height: 30px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; line-height: 40px;"><path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"></path></svg>
```
Comment (rupav) [inline src/popup.html]: Just align this as well. Scroll icon looks good to me now.

Comment (fragm3) [reply src/popup.html]: @rupav done

**Comment by fragm3** [comment]
@zamhaq resolved the conflicts

File: src/popup.html (hunk 1 above)
Comment (rupav) [reply src/popup.html]: I meant to say align the `svg` tag by 1 tab right, as it's inside the enclosing span tag.

**Comment by fragm3** [comment]
@zamhaq please merge it.

File: src/popup.html (hunk 1 above)
Comment (ms10398) [reply src/popup.html]: @fragm3 Can you make changes suggested by @rupav so that I can go ahead with the merge

File: src/script.js [hunk 2]
```diff
@@ -623,17 +622,30 @@ window.onload = function() {
...
 function handleScroll() {
-    var scrollIcon = scrollIconElement;
-    var end = messages.scrollHeight - messages.scrollTop === messages.clientHeight;
-    if (end) {
-        //hide icon
-        scrollIcon.style.display = "none";
-    } else {
-        //show icon
-        scrollIcon.style.display = "block";
+    var scrollIconTop = scrollIconTopElement;
```
Comment (ms10398) [inline src/script.js]: Use ES6

File: src/script.js [hunk 3]
```diff
@@ -11,7 +11,8 @@ var micimg = document.getElementById("micimg");
 var micmodal = document.getElementById("micmodal");
 var setting = document.getElementById("setting");
 var clear = document.getElementById("clear");
-var scrollIconElement = document.getElementById("scrollIcon");
+var scrollIconBottomElement = document.getElementById("scrollIconBottom");
```
Comment (ms10398) [inline src/script.js]: Use `let` and `const` instead of `var`

File: src/script.js [hunk 4]
```diff
@@ -869,9 +877,14 @@ function changeSpeak() {
     console.log("Should be speaking? " + shouldSpeak);
 }
 
+scrollIconTopElement.addEventListener("click", function(e) {
```
Comment (ms10398) [inline src/script.js]: Arrow Functions

**Comment by fragm3** [comment]
@ms10398 @rupav done the changes, please review

File: src/script.js [hunk 5]
```diff
@@ -646,17 +645,30 @@ window.onload = function() {
...
-    var end = messages.scrollHeight - messages.scrollTop === messages.clientHeight;
-    if (end) {
-        //hide icon
-        scrollIcon.style.display = "none";
-    } else {
-        //show icon
-        scrollIcon.style.display = "block";
+    let scrollIconTop = scrollIconTopElement;
+    let scrollIconBottom = scrollIconBottomElement;
+    var st = $(this).scrollTop();
```
Comment (ms10398) [inline src/script.js]: Please try to use ES6 wherever possible

File: src/script.js [hunk 6]
```diff
@@ -646,17 +645,30 @@ window.onload = function() {
...
+    scrollIconBottom.style.display = "none";
+    if (st > lastScrollTop){ // downscroll
+        if(messages.scrollHeight - messages.scrollTop === messages.clientHeight){
+            scrollIconTop.style.display = "block";
+        }
+        else {
+            scrollIconBottom.style.display = "block";
+        }
+    } else { // upscroll
+        if(messages.scrollTop === 0){
```
Comment (ms10398) [inline src/script.js]: Ternary Operator can be user here

File: src/script.js [hunk 7]
```diff
@@ -646,17 +645,30 @@ window.onload = function() {
...
-    } else {
-        //show icon
-        scrollIcon.style.display = "block";
+    let scrollIconTop = scrollIconTopElement;
+    let scrollIconBottom = scrollIconBottomElement;
+    var st = $(this).scrollTop();
+    scrollIconTop.style.display = "none";
+    scrollIconBottom.style.display = "none";
+    if (st > lastScrollTop){ // downscroll
+        if(messages.scrollHeight - messages.scrollTop === messages.clientHeight){
```
Comment (ms10398) [inline src/script.js]: Ternary Operator can be user here to simplify

**Comment by fragm3** [comment]
@ms10398 making the expressions ternary, gives travis error:
screenshot 2018-10-06 at 4 14 46 pm

I think if else is better since I am not getting a single value, (different assignment on diff condition)
Should I change it back to if else?

**Pull request closed without merging**